### PR TITLE
EOS-17755: Send and receive kv_payload as message

### PR DESCRIPTION
**Pull request by selvakumaar5496**
# Problem Statement
To test whether kv_payload can be sent and received through message_bus

# Design
https://jts.seagate.com/browse/EOS-17755

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [x] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [x] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [] Confirm Testing was performed with installed RPM [Y/N]: N (Not changing/ adding any interface)

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [x] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [x] Changes done to WIKI / Confluence page

### CODE REVIEW

**Review by sachinpunadikar — APPROVED** [review]
+1

File: py-utils/test/message_bus/test_kvpayload_message.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    _message_type = 'kv_payloads'
+    _kv_store = KvStoreFactory.get_instance('json:///etc/cortx/message_bus.conf')
+    _consumer = MessageConsumer(consumer_id='kv_consumer', consumer_group='kv', \
+        message_types=[_message_type], auto_ack=True, offset='earliest')
+    _producer = MessageProducer(producer_id='kv_producer', \
+        message_type=_message_type, method='sync')
+
+    def test_json_kv_send(self):
+        """ Load json as payload """
+        payload = TestKVPayloadMessage._kv_store.get_data()
```
Comment (ujjwalpl) [inline py-utils/test/message_bus/test_kvpayload_message.py:37]: payload need not come from the config file. Payload is any data in message. Example payload. 

```
payload = KvPayload{ 
    'attr1': 'val1',
    'attr2': 'val2', 
    ...
}
```

File: py-utils/test/message_bus/test_kvpayload_message.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import unittest
+from cortx.utils.kv_store import KvStoreFactory
+from cortx.utils.message_bus import MessageProducer, MessageConsumer
+
+
+class TestKVPayloadMessage(unittest.TestCase):
+    """ Test Send/Receive KvPayload as message """
+
+    _message_type = 'kv_payloads'
+    _kv_store = KvStoreFactory.get_instance('json:///etc/cortx/message_bus.conf')
```
Comment (ujjwalpl) [inline py-utils/test/message_bus/test_kvpayload_message.py:29]: There is no need to use kv_store for instantiating kv_payload. 
```
from kv_store.kv_payload import KvPayload
payload = KvPayload({....})
```

File: py-utils/test/message_bus/test_kvpayload_message.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    _consumer = MessageConsumer(consumer_id='kv_consumer', consumer_group='kv', \
+        message_types=[_message_type], auto_ack=True, offset='earliest')
+    _producer = MessageProducer(producer_id='kv_producer', \
+        message_type=_message_type, method='sync')
+
+    def test_json_kv_send(self):
+        """ Load json as payload """
+        payload = TestKVPayloadMessage._kv_store.get_data()
+        TestKVPayloadMessage._producer.send([json.dumps(payload)])
```
Comment (ujjwalpl) [inline py-utils/test/message_bus/test_kvpayload_message.py:38]: We need to send KvPayload as is without converting. 
```
producer.send(payload)
```

JSON could be a separate test and has nothing to do with KvPayload. Test name indicates that this is KvPayload test hence including JSON will be invalid or you can change the name of the test to say test_message_format.py

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/test/message_bus/test_kvpayload_message.py (hunk 2 above)
Comment (selvakumaar5496) [reply py-utils/test/message_bus/test_kvpayload_message.py:29]: done

File: py-utils/test/message_bus/test_kvpayload_message.py (hunk 1 above)
Comment (selvakumaar5496) [reply py-utils/test/message_bus/test_kvpayload_message.py:37]: done

File: py-utils/test/message_bus/test_kvpayload_message.py (hunk 3 above)
Comment (selvakumaar5496) [reply py-utils/test/message_bus/test_kvpayload_message.py:38]: MessageBus takes list of strings as input. KvPayload.get_data returns a dict. Hence converting it to sting obj and sending

Comment (selvakumaar5496) [reply py-utils/test/message_bus/test_kvpayload_message.py]: KvPayload.get_data(format_type='yaml') and KvPayload.get_data(format_type='toml') returns string hence no serialization required.

File: py-utils/test/message_bus/test_kvpayload_message.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    _payload = KvPayload({'message_broker': {'type': 'kafka', 'cluster': \
+        [{'server': 'localhost', 'port': '9092'}]}})
+    _consumer = MessageConsumer(consumer_id='kv_consumer', consumer_group='kv', \
+        message_types=[_message_type], auto_ack=True, offset='earliest')
+    _producer = MessageProducer(producer_id='kv_producer', \
+        message_type=_message_type, method='sync')
+
+    def test_json_kv_send(self):
+        """ Load json as payload """
+        message = TestKVPayloadMessage._payload.get_data()
```
Comment (ujjwalpl) [inline py-utils/test/message_bus/test_kvpayload_message.py:38]: We need to send payload itself. Not the data
message = Payload(....)
TestKVPayloadMessage._producer.send(message)

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

**Comment by selvakumaar5496** [comment]
Merged into single PR
https://github.com/Seagate/cortx-utils/pull/356

**Pull request closed without merging**